### PR TITLE
apply project ordering to timed assets just as it is applied to proximity assets

### PR DESCRIPTION
**Pull request by hburgund**
includes new unit test

### CODE REVIEW

**Comment by 13rac1** [comment]
Avoid copying/pasting code. Cut/paste is the best. Can you move this into a
new function instead?

**Comment by hburgund** [comment]
OK, so create a new function in recording_collection.py called something like "order_assets" and then call that new function from both places? Happy to do that; just making sure I understand your suggestion.

**Comment by 13rac1** [comment]
Yes. Exactly like that.

**Comment by hburgund** [comment]
@eosrei: check it out now; should be cleaner and more efficient. Seems to behave the same way.

File: roundwared/recording_collection.py [hunk 1]
```diff
@@ -184,6 +184,18 @@ def has_played(self):
         """
         return (len(self.banned_proximity) > 0 and len(self.banned_timeout) > 0)
 
+    def order_assets(self):
+        """
+        Order assets in playlist by ordering method set in project config
+        """
+        logger.debug("Ordering assets by: %s" % self.ordering)
+        if self.ordering == 'random':
+            self.playlist_timed = order_assets_randomly(self.playlist_timed)
```
Comment (13rac1) [inline roundwared/recording_collection.py]: This is only changing playlist_timed, not playlist_proximity. You need the function call to work like this:

```
    self.playlist_timed = self.order_assets(self.playlist_timed)
```

and

```
    self.playlist_proximity = self.order_assets(self.playlist_proximity)
```

**Comment by hburgund** [comment]
This is why different people always need to do the merge than the pull request ;-). Major oversight on my part; thanks for the catch. Fixed now...? No real reduction in code this way, but still better to consolidate similar functions and this improves readability, I think.
I can't think of a reason that it is bad to have both proximity and timed playlists re-ordered each time _either_ playlist is updated, can you?

**Comment by 13rac1** [comment]
Ah ha! No, I didn't mean like that. Take an asset list in the function and return just that list.
Like this:

```
    def order_assets(self, assets):
        """
        Order assets by ordering method set in project config
        """
        logger.debug("Ordering assets by: %s" % self.ordering)
        if self.ordering == 'random':
            return order_assets_randomly(assets)
        elif self.ordering == 'by_like':
            return order_assets_by_like(assets)
        elif self.ordering == 'by_weight':
            return order_assets_by_weight(assets)
```

**Comment by hburgund** [comment]
I'm not sure what's wrong here, but a stream unit test is now failing for some reason:

```
======================================================================
ERROR: test_modify_stream (tests.roundwared.test_stream.TestRoundStream)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vagrant/tests/roundwared/test_stream.py", line 54, in test_modify_stream
    stream.modify_stream(req)
  File "/vagrant/roundwared/stream.py", line 110, in modify_stream
    filenames = self.recordingCollection.get_filenames()
  File "/vagrant/roundwared/recording_collection.py", line 176, in get_filenames
    self.playlist_proximity)
TypeError: argument 2 to map() must support iteration

----------------------------------------------------------------------
```

Looks like my new ordering function somehow yields a `playlist_proximity` that is no longer iterable?
Oddly, I didn't get this failure with my original code duplication fix to this issue and I thought this new approach was essentially the same, but more efficient and sensible from a coding perspective.
Ideas?

**Comment by 13rac1** [comment]
OK! I figured it out. Previously in `recording_collection.py` :

``` python
        if self.ordering == 'random':
            self.playlist_proximity = order_assets_randomly(self.playlist_proximity)
        elif self.ordering == 'by_like':
            self.playlist_proximity = order_assets_by_like(self.playlist_proximity)
        elif self.ordering == 'by_weight':
            self.playlist_proximity = order_assets_by_weight(self.playlist_proximity)
```

Now it is:

``` python
self.playlist_proximity = self.order_assets(self.playlist_proximity)
```

with:

``` python
    def order_assets(self, assets):
        """
        Order assets by ordering method set in project config
        """
        logger.debug("Ordering assets by: %s" % self.ordering)
        if self.ordering == 'random':
            return order_assets_randomly(assets)
        elif self.ordering == 'by_like':
            return order_assets_by_like(assets)
        elif self.ordering == 'by_weight':
            return order_assets_by_weight(assets)
```

See the difference? It's super minor! In the original self.playlist_proximity is only changed if `self.ordering` is one of the required values. A reasonable assumption. But! In `test_stream.py` the ordering value is `1`. Yeah. That's not right. [Original code](https://github.com/roundware/roundware-server/blob/abce6b20522537c2898b7ec4b729d069314e9051/tests/roundwared/test_stream.py#L22) So `self.order_assets()` returns nothing, therefore `self.playlist_proximity` equals `None`.

So, we've discovered two issues. One, your new function should have a default return, so it's not setting `self.playlist_proximity` to `None`:

``` python
return []
```

Two, the `test_stream.py` needs to be fixed to use a valid ordering option.

``` python
        self.project1 = mommy.make(Project, name='Project One',
                                   ordering='random', recording_radius=16)
```

Additional note: You are outputting the "Ordering is" text twice. Line 189 and 228.

**Comment by hburgund** [comment]
aha! thank you, this is great sleuthing. I have made the updates and the tests now seem to pass, so I think we are good to go now, but definitely give it another once over to be sure.

File: roundwared/recording_collection.py [hunk 2]
```diff
@@ -184,6 +184,20 @@ def has_played(self):
...
+        Order assets by ordering method set in project config
+        """
+        logger.debug("Ordering assets by: %s" % self.ordering)
+        if self.ordering == 'random':
+            return order_assets_randomly(assets)
+        elif self.ordering == 'by_like':
+            return order_assets_by_like(assets)
+        elif self.ordering == 'by_weight':
+            return order_assets_by_weight(assets)
+        else:
```
Comment (13rac1) [inline roundwared/recording_collection.py]: Slightly pedantic, but you don't need this `else:`

**Pull request merged**